### ISSUE TITLE
Omitting or choosing SASS CSS Preprocessor causes "TypeError: Cannot read property 'plugins' of undefined" error

### ISSUE THREAD

**Issue by leighghunt** [kristianmandrup/generator-aurelia-ts#13]
I just noticed using the latest version. If I accept all the defaults, **_but omit selecting a processor, or select SASS**_ when asked to choose one, the following error occurs. Choosing Stylus results in the generator completing successfully.

```
test-aurelia-ts4 % yo
? 'Allo Leigh! What would you like to do? Aurelia Ts

Make sure you are in the directory you want to scaffold into.
This generator can also be run with: yo aurelia-ts

? application (package) name: test
? title: test
? description: test
? github account: leighghunt
? Your name: Leigh Hunt
? Your email: ###############
? Support IE9: Yes
? Decorate the app? Yes
Writing app files...
copying /build -> /build
copying /doc -> /docs
copying /test -> /test
   create package.json
   create README.md
   create .gitignore
   create index.html
   create aureliafile.js
   create .editorconfig
   create .jshintrc
   create npmignore
   create aurelia.protractor.js
   create gulpfile.js
   create favicon.ico
   create config.js
   create jsconfig.json
   create karma.conf.js
   create protractor.conf.js
   create LICENSE
? Install Styles: (Y/n) 
warn Running jspm globally, it is advisable to locally install jspm via npm install jspm --save-dev.
? Install Styles: Yes
? Install Aurelia CLI: Yes
? Install Aurelia Plugins: (y/N)      Looking up github:polymer/mutationobservers
     Updating registry cache...
ok   Installed polymer/mutationobservers as github:polymer/mutationobservers@^0.4.2 (0.4.2)
ok   Install tree has no forks.
     Looking up loader files...
       system.js
       system.src.js
       system-polyfills.js
       system-csp-production.js.map
       system-polyfills.js.map
       system-csp-production.js
       system-polyfills.src.js
       system.js.map
       system-csp-production.src.js

     Using loader versions:
       systemjs@0.18.9
     Looking up npm:babel-core
     Looking up npm:babel-runtime
     Looking up npm:core-js

warn Using local override for npm:core-js@0.9.18
ok   Installed babel as npm:babel-core@^5.8.20 (5.8.22)
     Looking up github:jspm/nodelibs-process
     Looking up github:jspm/nodelibs-fs
     Looking up github:systemjs/plugin-json
ok   Installed github:jspm/nodelibs-process@^0.1.0 (0.1.1)
     Looking up npm:process
ok   Installed github:jspm/nodelibs-fs@^0.1.0 (0.1.2)
ok   Installed github:systemjs/plugin-json@^0.1.0 (0.1.0)
ok   Installed npm:process@^0.10.0 (0.10.1)
ok   Installed babel-runtime as npm:babel-runtime@^5.8.20 (5.8.20)
ok   Installed core-js as npm:core-js@^0.9.4 (0.9.18)
ok   Loader files downloaded successfully

ok   Install complete.
? Install Aurelia Plugins: No
? Install TypeScript: No
? Visual Studio: No
? Install UI Frameworks: Yes
? CSS Preprocessors:                              // <=== either leave blank or choose SASS
? Remove old styles? No
? Use Jade Templates? No
? Font Awesome: Yes
? Layout frameworks: Bootstrap
? Primary layout framework: Bootstrap
? Install CLI in global registry: No
? Ampersand models and collections: Yes
copying /src -> /src
copying /styles/sass -> /styles/sass
copying /styles/stylus -> /styles/stylus
copying /styles/css -> /styles/css

     _-----_
    |       |    .----------------------.
    |--(o)--|    |     Bye from us!     |
   `---------´   |      Chat soon.      |
    ( _´U`_ )    |                      |
    /___A___\    |      Yeoman team     |
     |  ~  |     |   http://yeoman.io   |
   __'.___.'__   '----------------------'
 ´   `  |° ´ Y ` 

events.js:85
      throw er; // Unhandled 'error' event
            ^
TypeError: Cannot read property 'plugins' of undefined
    at styles (/Users/lhunt/.npm-packages/lib/node_modules/generator-aurelia-ts/generators/styles/write.js:25:86)
    at /Users/lhunt/.npm-packages/lib/node_modules/generator-aurelia-ts/lib/writer.js:12:28
    at Object.writeAll (/Users/lhunt/.npm-packages/lib/node_modules/generator-aurelia-ts/lib/writer.js:19:24)
    at module.exports.yeoman.generators.Base.extend.writing (/Users/lhunt/.npm-packages/lib/node_modules/generator-aurelia-ts/generators/styles/index.js:83:17)
    at /Users/lhunt/.npm-packages/lib/node_modules/generator-aurelia-ts/node_modules/yeoman-generator/lib/base.js:421:16
    at processImmediate [as _immediateCallback] (timers.js:367:17)
test-aurelia-ts4 % 
```

**Comment by kristianmandrup** [comment]
Yup! I never got around to support SASS. I completely abandoned SASS, Stylus is WAY better and more much more powerful IMO. If you really need/want SASS for whatever reason, send me a Pull Request ;)
Cheers!

**Comment by leighghunt** [comment]
I don't need SASS myself - but thought I'd log the issue and workaround whilst I was exploring and it was in my mind, so others could find it! :smiley: 

Cheers, and thanks for your work!
Leigh.

**Comment by kristianmandrup** [comment]
I will add SASS shortly, just for the sake of it...

**Comment by kristianmandrup** [comment]
Published a new release `0.7.3` which fixes this, now full SASS support!
See Changelog for other improvements (prompt defaults and aurelia install via registry lookup!!)

**Comment by telekosmos** [comment]
Hi0.7.3 Cool! Mind two tests are failing. Anyway, I am refactoring the tests and checking another issues:- did you take --options into account? I ran some tests and it seems if you set a --option, the related prompt is still 'prompting', at least in styles generator.- as we include SCSS support, we can include SASS support (my preferred choice :-)- regarding to layout generator options --ui xxx, is this intended as the primary layout? may I include several, lets say, --ui f7 zurb? As before, prompts are still displaying even if you set the option and prompts override it

First I am trying to run the tests based on the documentation, then I check the index.js in order to tests get passed as the generator is now; then I ask in order not to misunderstand :-). However I updated the styles generator to:- options gets priority over prompts- tricky issue: I couldn't run the generator in two different suites because of the install = install(this); line in the initializing stage and I replace that line by this.install = install(this). Tests pass, it seems it does not affect the generator but if you think or find out is wrong just let me know

Cheers!! -- Guillermo C. Martínez 

```
 On Saturday, August 15, 2015 10:59 AM, Kristian Mandrup <notifications@github.com> wrote:
```

 Published a new release 0.7.3 which fixes this, now full SASS support!
See Changelog for other improvements (prompt defaults and aurelia install via registry lookup!!)—

**Comment by kristianmandrup** [comment]
Hi Guil,

Yeah, I’m aware there seems to be a problem with taking in the application argument and options. Looks like the prompt is instantiated before the options are!?
Why I had to move the require(‘./defaults’) inside the prompts.js file for the app generator, since if I loaded defaults in initialiser, the prompt would think it was still undefined!

Yes,  --ui is an “old” option. Should support a list of ui’s as you suggest! You can also improve the install vs this.install. Is very confusing. Perhaps always use the lib.install if it comes from lib. Thanks :) Looking forward to see more improvements!! Happy that you like the new SASS integration. You are welcome to add Compass, Bourbon etc. of course :)
Cheers!

Kris
